internal/app/grpcclient/impl.go

```diff
@@ -113,24 +115,224 @@ func (i *invoker) unary(
 }
 
 func (i *invoker) serverStream(
-	_ context.Context,
-	_ *v1alpha1.ClientCompatRequest,
+	ctx context.Context,
+	ccr *v1alpha1.ClientCompatRequest,
 ) (*v1alpha1.ClientResponseResult, error) {
-	return nil, errors.New("server streaming is not yet implemented")
+	msg := ccr.RequestMessages[0]
+	req := &v1alpha1.ServerStreamRequest{}
+	if err := msg.UnmarshalTo(req); err != nil {
+		return nil, err
+	}
+
+	// Add the specified request headers to the request
+	ctx = grpcutil.AppendToOutgoingContext(ctx, ccr.RequestHeaders)
+
+	stream, err := i.client.ServerStream(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+	// Read headers and trailers from the stream
+	hdr, err := stream.Header()
+	if err != nil {
+		return nil, err
+	}
+	headers := grpcutil.ConvertMetadataToProtoHeader(hdr)
+	trailers := grpcutil.ConvertMetadataToProtoHeader(stream.Trailer())
```

Comment:
You can't call `stream.Trailer` until you've reached the end of the messages. So you need to move this below the `for` `stream.Recv` loop.